app/models/activity.rb

```diff
@@ -0,0 +1,62 @@
+# Activity model inherited from public_activity gem is used to customize activities' management
+class Activity < PublicActivity::Activity
+  acts_as_readable on: :created_at
+  after_create :send_email
+
+  # The formats of notification that are supported
+  SUPPORTED_FORMATS = [:feed, :email, :popup]
+
+  # Class methods used to create an activity
+  class << self
+    # Create an activity record and store the notification formats of it
+    #
+    # @param from the one triggers this activity
+    # @param [User] to the user who will receive notifications
+    # @param trackable the object that will be tracked in this activity
+    # @param [Symbol] event the name of the activity indicating which event happened
+    # @param [Array<Symbol>] formats the formats of the notifications that will be used to inform
+    #   users. Any combination of :activity_feed, :email and :popup, is valid
+    def create(from, to, trackable, event, formats: [])
```

Comment:
I think a more sane default for the formats would be good. Also allow users to specify only one thing without an array.